src/auth.nim

```diff
@@ -227,20 +198,45 @@ proc initAccountPool*(cfg: Config; path: string) =
   elif fileExists(path):
     log "Parsing JSON guest accounts file: ", path
     accountPool = parseGuestAccounts(path)
-  elif guestAccountsUrl == "":
-    echo "[accounts] ERROR: ", path, " not found. This file is required to authenticate API requests."
+  elif not cfg.guestAccountsUsePool:
+    echo "[accounts] ERROR: ", path, " not found. This file is required to authenticate API requests. Alternatively, configure the guest account pool in nitter.conf"
     quit 1
 
-  let accountsPrePurge = accountPool.len
   accountPool.keepItIf(not it.hasExpired)
 
   log "Successfully added ", accountPool.len, " valid accounts."
 
+proc updateAccountPool*(cfg: Config) {.async.} =
+  if not cfg.guestAccountsUsePool:
+    return
+
+  while true:
+    if accountPool.len == 0:
+      let pool = HttpPool()
```

Comment:
What I meant is that it's quite overkill to use a http pool here, as it keeps open connections sitting around while you don't need them in this case. It should probably be moved to another file, but you can use `safeFetch` from media.nim for now as a simple and easy way to just get the contents without caring about handling the http client or errors.

Reply:
i don't think i want that function. I still want to log status code and log exceptions. should I just use the stdlib directly?

Reply:
Sure, that makes sense. Just make sure to do `client.close()` in a `final:` clause.